components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolverTest.java

```diff
@@ -188,6 +161,40 @@ public void testGetInstance() {
         assertEquals(identityKeyStoreResolver1, identityKeyStoreResolver2);
     }
 
+    @DataProvider(name = "MissingConfigDataProvider")
+    public String[] missingConfigDataProvider() {
+
+        return new String[] {
+                "identity_err1.xml",
+                "identity_err2.xml",
+                "identity.xml" // Finally loading correct identity.xml file to be used for other tests.
+        };
+    }
+
+    @Test(dataProvider = "MissingConfigDataProvider")
+    public void testMissingConfigs(String fileName) {
+
+        try {
+            // Set instance to null for creating a new instance
+            Field identityKeyStoreResolverInstance = IdentityKeyStoreResolver.class.getDeclaredField("instance");
+            identityKeyStoreResolverInstance.setAccessible(true);
+            identityKeyStoreResolverInstance.set(null, null);
+
+            // Use custom identity.xml file from test resources.
+            String identityXmlPath = Paths.get(System.getProperty("user.dir"), "src", "test", "resources",
+                    fileName).toString();
+
+            // Mock IdentityConfigParser.
+            mockIdentityConfigParser = IdentityConfigParser.getInstance(identityXmlPath);
+            identityConfigParser.when(IdentityConfigParser::getInstance).thenReturn(mockIdentityConfigParser);
+
+            // Test instance creation --> Config read.
+            IdentityKeyStoreResolver.getInstance();
+        } catch (Exception e) {
+            fail();
```

Comment:
Please add the exception details in the fail()
ex:     `fail("Test failed due to exception: " + e.getMessage());`